modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -191,8 +201,26 @@ def _read_csv_check_support(
         if read_csv_kwargs.get("chunksize") is not None:
             return False
 
-        skiprows = read_csv_kwargs.get("skiprows")
-        if skiprows is not None and not isinstance(skiprows, int):
+        dummy_df = next(
+            pandas.read_csv(
+                filepath_or_buffer,
+                **dict(
+                    read_csv_kwargs,
+                    skiprows=None,
+                    skipfooter=0,
+                    usecols=None,
```

Comment:
What if `usecols != None`?

Reply:
If we use `usecols` parameter we will return the number of columns in the resulted data frame, not the real number of columns that data file contain.

Reply:
I mean, if the user is going to use this option and he only needs half of the columns from the file, how will that affect your empirical rule of 32 columns?

Reply:
Even in the case when user needs only part of columns, we still have to process all columns of data file (`_read_rows` function processes whole line with no respect to `usecols` parameter) and `usecols` will be used by parsers by workers. So in that case, in my opinion, we don't need to check this parameter.